### PR TITLE
Feature: At mention submit

### PR DESCRIPTION
**Pull request by bzang**
Allow user to submit at mentions. Currently on allows people in the same room.

### CODE REVIEW

**Comment by adamweeks** [comment]
@bzang could you throw some screen shots of the @ mention popup?

**Comment by bzang** [comment]
Sure thing. @adamweeks 
initial
search
final

File: packages/node_modules/@ciscospark/react-container-message-composer/src/container.js [hunk 1]
```diff
@@ -103,49 +115,128 @@ export class MessageComposer extends Component {
...
+      }
+    });
+
+    if (term) {
+      return participants.filter((p) => p.display.toLowerCase().includes(term.toLowerCase()));
+    }
+    return participants;
+  }
+
+  @autobind
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/react-container-message-composer/src/container.js]: I don't think we need `@autobind` here, do we?

File: packages/node_modules/@ciscospark/widget-message/src/container.js [hunk 2]
```diff
@@ -100,10 +100,15 @@ export class MessageWidget extends Component {
...
   componentWillUpdate(nextProps) {
     const props = this.props;
+
     if (this.activityList
       && nextProps.conversation.get(`activities`).count() !== props.conversation.get(`activities`).count()) {
       this.scrollHeight = this.activityList.getScrollHeight();
     }
+
+    if (!nextProps.features.hasIn([`items`, `js-widgets-mentions`])) {
+      props.getFeature(`developer`, `js-widgets-mentions`, nextProps.sparkInstance);
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/widget-message/src/container.js]: `getFeature` isn't imported or in proptypes

File: packages/node_modules/@ciscospark/react-container-message-composer/src/container.js (hunk 1 above)
Comment (bzang) [reply packages/node_modules/@ciscospark/react-container-message-composer/src/container.js]: Good catch

File: packages/node_modules/@ciscospark/widget-message/src/container.js (hunk 2 above)
Comment (bzang) [reply packages/node_modules/@ciscospark/widget-message/src/container.js]: Oh wow, guess `getFeature` was getting passed by the container. I'll add that into this widget so we don't run into odd issues.

**Comment by bzang** [comment]
@adamweeks, fixed. Ready for another round

File: packages/node_modules/@ciscospark/widget-message/src/container.js [hunk 3]
```diff
@@ -736,6 +742,7 @@ export default connect(
     setTyping,
     showScrollToBottomButton,
     updateHasNewMessage,
-    updateWidgetState
+    updateWidgetState,
+    getFeature
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/widget-message/src/container.js]: Can you add it to the injected prop types as well?

Comment (bzang) [reply packages/node_modules/@ciscospark/widget-message/src/container.js]: Yup. Done

File: packages/node_modules/@ciscospark/widget-message/src/container.js [hunk 4]
```diff
@@ -700,7 +700,8 @@ const injectedPropTypes = {
   updateHasNewMessage: PropTypes.func.isRequired,
   updateWidgetState: PropTypes.func.isRequired,
   space: PropTypes.object.isRequired,
-  features: PropTypes.object.isRequired
+  features: PropTypes.object.isRequired,
+  getFeature: PropTypes.object.isRequired
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/widget-message/src/container.js]: `getFeature` is a function 😭

**Pull request merged**